WalletWasabi/Blockchain/Keys/HdPubKeyCache.cs

```diff
@@ -1,31 +1,28 @@
 using System.Collections;
 using System.Collections.Generic;
-using System.Collections.Immutable;
 using System.Diagnostics.CodeAnalysis;
 using System.Linq;
 using NBitcoin;
 
 namespace WalletWasabi.Blockchain.Keys;
 
-public class HdPubKeyCache : IEnumerable<HdPubKey>
+public class HdPubKeyCache : IEnumerable<HdPubKeyInfo>
```

Comment:
Whatever we modify in the code has real consequences observable from the external world, otherwise why would someone to modify code? This means that this should be observable through the `KeyManager`. You should be able to request the keys, generate a new key and verify that the enumeration that you received before adding the key doesn't contain it.

Reply:
Sure. You are testing exactly as I suggested (testing the KeyManager). So, yes, except for the code added for testing only, which is not necessary.

The `SameKeyTwiceTest` is not good because it assumes the cache must fail in case of duplicated keys but that is not a responsibility of the cache. In case the cache receives a duplicated key that means that we have a bug in the KeyManager and we need to fix it.